src/PrestaShopBundle/Form/Admin/Type/TranslateTextType.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Class is responsible for creating translatable text inputs
+ */
+class TranslateTextType extends AbstractType
```

Comment:
Why did you make a new FormType independent from the existing `PrestaShopBundle\Form\Admin\Type\TranslateType`?
They seem to be quite related, but they do not share any code.

Reply:
Hmmm, thats a good point @Quetzacoalt91. Is it used to render textareas? 
Maybe it to would be the best to have TranslateTextType & TranslateTextareaType which shares common code?

Reply:
Not exactly. It seems to display the language selector. However it must be combined with the Type you want to display:
### Text:
https://github.com/PrestaShop/PrestaShop/blob/3f9808dd5b53c75529ec9243f00bdba9857cc594/src/PrestaShopBundle/Form/Admin/Product/ProductInformation.php#L143-L158

### TextArea with TinyMce (prior #8604 ):
https://github.com/PrestaShop/PrestaShop/blob/3f9808dd5b53c75529ec9243f00bdba9857cc594/src/PrestaShopBundle/Form/Admin/Product/ProductInformation.php#L159-L177

### TextArea with TinyMce (from #8604):
https://github.com/Quetzacoalt91/PrestaShop/blob/8a5f67d5e2651de8a15e3a875fc51707cd0496cd/src/PrestaShopBundle/Form/Admin/ShopParameters/General/MaintenanceType.php#L58-L66

It seems you just had to adapt the existing template in order to display the selector properly.
However, this requires some check on the Product page, where these inputs already exist.

Reply:
ok, thanks for clarification. :) However, I'm no sure if it's a good idea to have 'all in one' type. Problem I see is that you have to specify which type you want to combine, does this mean you can combine ChoiceType, DateTimeType or any other type available in Symfony? It's confusing. I would like to see different types for different inputs, like Symfony does it with TextType, TextareaType and others so you can add it to your form without any addtional options. For us it means we dont have to adapt TranslateType template for many different cases. It's just my thoughts. What do you think @Quetzacoalt91?

Reply:
I haven't tested, but I do not see why we could not use the other Symfony types.
As you can see on the 2 first lines, we always have the choice to use the input type we want.

... or I misunderstood what you meant?

Reply:
yeah, my idea was a bit different.

do you think i should adapt TranslateType and use it in this page?

Reply:
Basically we have a duplication of code here between `PrestaShopBundle\Form\Admin\Type\TranslateType` && `PrestaShopBundle\Form\Admin\Type\TranslateTextType`.

Let's keep your solution as well as you made a specific integration for this page. But we **will have to** merge these classes prior the 1.7.4.0 release. The objective is to have the same behavior everywhere. If we need to adapt the existing pages, we will.

Reply:
Can be done on another PR. The task has been created: http://forge.prestashop.com/browse/BOGOSS-87

Reply:
thanks @Quetzacoalt91. Unfortunately I cannot see the task, I get 404 even though i'm logged in.